### DISCUSSION TITLE
Option to use eloquent instead of db query builder for validation rule "exists"

### DISCUSSION THREAD

**Discussion by Sairahcaz** [laravel/framework#49831] [Ideas]
It would be very cool to have an option, to make the validation rule "exists" use the eloquent query builder, instead of the db query builder. This way your global scopes would get applied.

In a very simple multi-tenancy implementation, imagine you have the following model:

```php
class Project extends Model
{
    use HasTenancyScope;
    ....
}
```

```php
trait HasTenancyScope
{
    public static function bootHasTenancyScope(): void
    {
        parent::addGlobalScope('tenancy', function (Builder $query) {
            if (auth()->check()) {
                $query->where('tenancy_id', auth()->user()->tenancy_id);
            }
        });

        static::saving(function ($model) {
            if (auth()->check()) {
                $model->tenancy_id = auth()->user()->tenancy_id;
            }
        });
    }
}
```

Now if you try to validate a project_id, you always need to make sure to not forget the tenancy_id check:
```php
public function rules(): array
    {
        return [
            'projectId' => [
                'required',
                'integer',
                Rule::exists(Project::class, 'id')->where(function ($query) {
                    $query->where('tenancy_id', auth()->user()->tenancy_id);
                    $query->where('id', $this->projectId);
                }),
            ],
    }
```

This is pretty cumbersome and error-prone. 
It would be super convenient to have an option to make the validation rule internally using the eloquent query builder, so that the global scopes get applied! Maybe this could be done if you specifically use the model namespace validation format:
```
'project_id' => 'exists:App\Models\Project,id'
```

But for sure it could also be done differently, if you do not want to introduce a breaking change here.

I would love to create a PR for this, if there is any chance it get merged?

**Comment by MrPunyapal** [comment]
It's cool 👌

**Comment by tpetry** [comment]
Could have used this in the past too - especially because of soft-deleted. You could create a new `Rule::modelExists()` as changing the current behaviour would break a ton of applications. Or maybe just a new parameter `bool $scopes` which is by default false. 

Would be very easy to distinguish with named parameters:

```php
Rule::exists(Project::class, 'id', scopes: true),
```

**Reply by Sairahcaz** [reply]
Yeah `scopes: true` would be an option.

Though I still feel like this should be the default behavior, especially if you exactly use your model and not a table name.
This just feels more intuitive to me...

Maybe something for Laravel 11? ^^

**Comment by ArielMejiaDev** [comment]
I think that Validator is a macroable class, so in case the idea is no accepted, you could add your own ‘existsInCurrentTenant’ rule
